### PR TITLE
Mw/release scripting

### PR DESCRIPTION
**Pull request by wilkermichael**
Changes proposed in this PR:
- Modified scripts and created `prepare-release` and `prepare-dev` functions to be used in the makefile
- Added a new make target `make prepare-dev`
- Removed old `dev.sh` file and `set_dev_mode` function
- When running `dev.sh` it performs the git actions for you, I decided to make the creation of a PR a mandatory step to pair better with what is done in the existing `prepare-release` make target

## USAGE
1. Dev starts the release process
2. Dev updates changelog with all necessary changes
3. Dev creates a new release branch, for example `release/0.49.0`
4. Dev runs `make prepare-release` on `release/0.49.0` branch
5. Dev pushes changes
6. Dev completes release
7. Dev checks out `main`
8. Dev runs `make prepare-dev` and opens a PR on main

- I've included three commits which reflect the file states after `make prepare-release`, `make prepare-dev` and `make prepare-dev` when using a pre-release tag. I will drop these after they are reviewed, but for ease of review examine these two commits for the proposed results of running these make targets.

How I've tested this PR:
- ran make targets locally

How I expect reviewers to test this PR:
- run make targets locally

Checklist:
- [N/A] Tests added
- [N/A] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
-## 0.49.0 (September 28, 2022)
+## UNRELEASED
+
+## 0.49.0-dev (September 28, 2022)
```
Comment (wilkermichael) [inline CHANGELOG.md:3]: This is mistake, I'll fix this

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,5 @@
+## UNRELEASED
+
 ## 0.49.0 (September 28, 2022)
```
Comment (wilkermichael) [inline CHANGELOG.md:3]: This is included in the dev changes

**Review by curtbushko — COMMENTED** [review]
Looks pretty good!

I only have 1 tiny nit but nothing blocking.

File: Makefile [hunk 3]
```diff
@@ -143,13 +143,27 @@ prepare-release: ## Sets the versions, updates changelog to prepare this reposit
 ifndef RELEASE_VERSION
 	$(error RELEASE_VERSION is required)
 endif
-	source $(CURDIR)/control-plane/build-support/scripts/functions.sh; set_release_mode $(CURDIR) $(RELEASE_VERSION) "$(shell date +"%B %d, %Y")" $(PRERELEASE_VERSION)
+ifndef RELEASE_DATE
+    $(error RELEASE_DATE is required)
```
[the remark this answers is not in the document]
Comment (wilkermichael) [reply Makefile:147]: That's a really good add, thanks for bringing it up. I'll add this

File: charts/consul/Chart.yaml [hunk 4]
```diff
@@ -10,12 +10,12 @@ sources:
   - https://github.com/hashicorp/consul
   - https://github.com/hashicorp/consul-k8s
 annotations:
-  artifacthub.io/prerelease: false
+  artifacthub.io/prerelease: true
   artifacthub.io/images: |
     - name: consul
       image: hashicorp/consul:1.13.1
```
Comment (kschoche) [inline charts/consul/Chart.yaml:16]:
```suggestion
      image: hashicorp/consul:1.13.2
```

**Review by kschoche — APPROVED** [review]
🔥

File: charts/consul/Chart.yaml (hunk 4 above)
Comment (wilkermichael) [reply charts/consul/Chart.yaml:16]: ah yes, this isn't part of the automation. This step is done manually atm, it'll make more sense with the release process revision coming to the engineering docs soon!

**Review by curtbushko — APPROVED** [review]

**Pull request merged**